growthbook/growthbook
# Updates RunQueries permission checks on front end

### Features and Changes
While doing some testing, I found that we weren't doing any permission checks before trying to generate or re-generate an information schema for a particular datasource, resulting in not only permission errors, but permission errors that weren't being handled and causing the app to crash.

Now, we are checking for permission on the frontend before we attempt to generate/regenerate a datasource's information schema.

Additionally, we didn't have any protections around the `Test Queries` permissions - now, in the EditSQLModal we are disabling the `Test Query` button and adding a tooltip if the user doesn't have 
 `canRunTestQueries` permission.

Likewise, we are automatically unchecking the `Test queries before saving` if the user doesn't have test query permissions, and disabling the input + wrapping it in a tooltip.

Reproduction steps / Testing
- Create a custom role with the following policies - `DatasourcesFullAccess`, `DataSourceConfiguration`, and `MetricsFullAccess`.
- Apply this role to a user and then attempt to create a metric on a datasource that supports information_schemas
- Click the `Edit SQL` button and see that there is no information schema or the info schema for the connected datasource is out of date and needs to be refreshed.
- Ensure that the CTA is disabled with a tooltip.

### Screenshots
Screenshot of the info schema table refresh button when user doesn't have permission
<img width="454" alt="Screen Shot 2024-05-20 at 9 13 16 AM" src="https://github.com/growthbook/growthbook/assets/75274610/b3d81ba2-e433-4290-adc1-63569090fb90">

Screenshot of the info schema databaes refresh button when user doesn't have permission
<img width="463" alt="Screen Shot 2024-05-20 at 9 13 09 AM" src="https://github.com/growthbook/growthbook/assets/75274610/7da3c840-6733-457c-b7c2-54070d0d09c3">

Screenshot of the disabled input when user doesn't have test queries permission
<img width="520" alt="Screen Shot 2024-05-20 at 9 12 29 AM" src="https://github.com/growthbook/growthbook/assets/75274610/5013afcb-aa11-4613-83bd-e51d5076173e">

Screenshot of the disabled Test Query button if the user doesn't have test queries permission
<img width="879" alt="Screen Shot 2024-05-20 at 9 08 24 AM" src="https://github.com/growthbook/growthbook/assets/75274610/ea5f624f-e60c-4998-9598-e14c8cc1640a">

Screenshot of the disabled retry button when the user doesn't have canRunSchemaQueries permission
<img width="423" alt="Screen Shot 2024-05-20 at 9 08 28 AM" src="https://github.com/growthbook/growthbook/assets/75274610/fb8432a5-076e-44f5-9282-04b77552d5fe">
